src/App.jsx

```diff
@@ -1,7 +1,101 @@
+import { useState } from "react";
 import "./App.css";
+import BottomModalLayout from "./components/BottomModalLayout/BottomModalLayout";
+import Header from "./components/common/Header/Header";
+import RestaurantListLayout from "./components/RestaurantListLayout/RestaurantListLayout";
+import RestaurantDetailModal from "./components/BottomModalLayout/RestaurantDetailModal/RestaurantDetailModal";
+import AddRestaurantModal from "./components/BottomModalLayout/AddRestaurantModal/AddRestaurantModal";
+import { useEffect } from "react";
 
 function App() {
-  return <h1>Self-Paced React</h1>;
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalName, setModalName] = useState("");
```

Comment:
모달 open, close 상태와 어떤 모달이 렌더링될 지 결정하는 이름의 상태를 합치면 어떨까요??
```suggestion
  const [modalState, setModalState] = useState({
    isOpen: false,
    name: "",
  });
```
어차피 두 상태는 서로 의존적이여서 하나로 관리하는게 좋을 것 같은데 카멜의 생각은 어떠신가요??🙃

Reply:
호오...? 좋은데요...? 뭔가 모달 관련된 상태가 써밋 말대로 어차피 의존적이라 하나로 묶고싶었는데 객체 상태로 관리하면 되겠군요!👍

Reply:
좋은데요~?